### PR TITLE
Fix #6559

### LINKED ISSUE #6559: wifi.radio.tx_power does not support float
The [Wemos docs say to use 8.5db](https://www.wemos.cc/en/latest/c3/c3_mini.html#about-wifi) to work around the WIFI connect issue.

## Workaround

If you use 8 or 9 you can still connect.

### Code/REPL

```python
>>> import wifi
>>> wifi.radio.tx_power = 8.5
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't convert float to int
>>> wifi.radio.tx_power = 8
>>> wifi.radio.connect(SSID, PASSWD)
>>> wifi.radio.ipv4_address
192.168.1.53
```

### Behavior

TypeError: can't convert float to int

### Description

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by todbot**
This API should be float instead of int.

### CODE REVIEW

**Review by askpatrickw — APPROVED** [review]
Built locally and tested and it worked for me.
Thanks @todbot  !!

File: ports/espressif/common-hal/wifi/Radio.c [hunk 1]
```diff
@@ -139,14 +139,14 @@ void common_hal_wifi_radio_set_mac_address(wifi_radio_obj_t *self, const uint8_t
     esp_wifi_set_mac(ESP_IF_WIFI_STA, mac);
 }
 
-uint8_t common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self) {
+float common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self) {
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:142]:
```suggestion
mp_float_t common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self) {
```

File: ports/espressif/common-hal/wifi/Radio.c [hunk 2]
```diff
@@ -139,14 +139,14 @@ void common_hal_wifi_radio_set_mac_address(wifi_radio_obj_t *self, const uint8_t
...
     int8_t tx_power;
     esp_wifi_get_max_tx_power(&tx_power);
-    return tx_power / 4;
+    return tx_power / 4.0;
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:145]:
```suggestion
    return tx_power / 4.0f;
```

File: ports/espressif/common-hal/wifi/Radio.c [hunk 3]
```diff
@@ -139,14 +139,14 @@ void common_hal_wifi_radio_set_mac_address(wifi_radio_obj_t *self, const uint8_t
...
 }
 
-void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const uint8_t tx_power) {
-    esp_wifi_set_max_tx_power(tx_power * 4);
+void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const float tx_power) {
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:148]:
```suggestion
void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const mp_float_t tx_power) {
```

File: ports/espressif/common-hal/wifi/Radio.c [hunk 4]
```diff
@@ -139,14 +139,14 @@ void common_hal_wifi_radio_set_mac_address(wifi_radio_obj_t *self, const uint8_t
...
+    esp_wifi_set_max_tx_power(tx_power * 4.0);
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:149]:
```suggestion
    esp_wifi_set_max_tx_power(tx_power * 4.0f);
```

File: shared-bindings/wifi/Radio.h [hunk 5]
```diff
@@ -82,8 +82,8 @@ extern void common_hal_wifi_radio_set_mac_address(wifi_radio_obj_t *self, const
 extern mp_obj_t common_hal_wifi_radio_get_mac_address_ap(wifi_radio_obj_t *self);
 extern void common_hal_wifi_radio_set_mac_address_ap(wifi_radio_obj_t *self, const uint8_t *mac);
 
-extern uint8_t common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self);
-extern void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const uint8_t power);
+extern float common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self);
+extern void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const float power);
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.h:86]:
```suggestion
extern mp_float_t common_hal_wifi_radio_get_tx_power(wifi_radio_obj_t *self);
extern void common_hal_wifi_radio_set_tx_power(wifi_radio_obj_t *self, const mp_float_t power);
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks! Minor changes here: use `mp_float_t` to pass floating-point values around between `shared-bindings` and `common-hal`. `mp_float_t` is defined to be `float` or `double`, depending on other implementation choices. This is the same idea as the`mp_int_t` type.

File: ports/espressif/common-hal/wifi/Radio.c (hunk 2 above)
Comment (jepler) [reply ports/espressif/common-hal/wifi/Radio.c:145]: `MICROPY_FLOAT_CONST(4.0)` is even more technically correct, even though we don't correctly follow that style everywhere (yet).

Comment (jepler) [reply ports/espressif/common-hal/wifi/Radio.c:145]: Similar to how `mp_float_t` is either `float` or `double`, `MICROPY_FLOAT_CONST` appends the `f` to the literal constant (or not) depending on the actual type of `mp_float_t`.

**Review by tannewt — APPROVED** [review]

**Comment by tannewt** [comment]
> Are these other functions I've seen using an older philosophy or, in a perfect world, should those be changed too?

I'm lazy sometimes and don't use the mp_ versions even when they are generally better to use.

**Pull request merged**